DatadogInternal/Sources/NetworkInstrumentation/URLSession/DatadogURLSessionDelegate.swift

```diff
@@ -19,6 +21,7 @@ public protocol __URLSessionDelegateProviding: URLSessionDelegate {
 ///
 /// All requests made with the `URLSession` instrumented with this delegate will be intercepted by the SDK.
 @objc
+@available(*, deprecated, message: "Use URLSessionInstrumentation instead.")
```

Comment:
suggestion/ Most users will face this warning without much context (seeing docs or migration guide), so I wonder if it is informative enough. _What is URLSessionInstrumentation? Is this a module? A class?_

IMO, better to reference entry-point API, so users could search for it and try to figure-it out:
```swift
@available(*, deprecated, message: "Use `URLSessionInstrumentation.enable(with:)` instead.")
```

However, best IMO would be to enhance this public API comment to better explain the usage of `URLSessionInstrumentation`. Then, calling this symbol from deprecation warning could be enough 👌.
https://github.com/DataDog/dd-sdk-ios/blob/643d26e6ba2ce6582f28aee2a1789f6dccc94491/DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionInstrumentation.swift#L9-L10

Reply:
that's fair call, I just made the changes.